src/main/java/io/jenkins/plugins/intotorecorder/InTotoWrapper.java

```diff
@@ -186,12 +222,118 @@ public void setUp(SimpleBuildWrapper.Context context,
         this.link = new Link(null, null, this.stepName, null, null, null);
         this.link.setMaterials(InTotoWrapper.collectArtifacts(this.cwd));
 
+        //Setting up Provenance object
+        this.provenance = new Provenance();
+
+        //Setting up the builder
+        setBuilder(initialEnvironment);
+
+        //setting up materials
+        setProvenanceMaterials(InTotoWrapper.collectArtifacts(this.cwd));
+
+        //setting up Invocation
+        this.invocation = new BuildInvocation();
```

Comment:
Why can't this be locally scoped?

Reply:
I'm guessing it's because of `setProvenanceMaterials` etc but I suspect it can be cleaner.

Reply:
Are you referring to the `BuildInvocation` object? 
I think there is no need as such for it to be global, we can make it local as well.

Reply:
Yep, I think I agree. The same probably goes for the `metadata` object as well?

Reply:
Yes, I'll do that